### DISCUSSION TITLE
Issues with Firebase AppCheck Blocking Firestore Requests in iOS App Built with Codemagic

### DISCUSSION THREAD

**Discussion by FltSv** [codemagic-ci-cd/codemagic-docs#2700] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Use Codemagic to build a release version of a Flutter project for iOS.
1. The Flutter project uses Firebase Firestore and has Firebase AppCheck enabled for security.
1. Deploy the app to TestFlight and install it on an iPhone device.
1. On the iPhone, launch the app and attempt to access Firestore data.
- I've followed the setup as per Firebase and Codemagic documentation.
- Tried activating FirebaseAppCheck using both appAttest and deviceCheck as the appleProvider in FirebaseAppCheck.instance.activate(appleProvider: ).
- Attempted iOS code signing with both Automatic and Manual options in Codemagic.

### Expected results

- The app should successfully launch on the iPhone.
- Firestore access should work without issues, with no blocks from Firebase AppCheck.

### Actual results

- The app launches without any apparent issues.
- However, when trying to access Firestore, the request is blocked.
- Firebase Console logs show the request as "Invalid Request," and it is being blocked by AppCheck.
- Disabling AppCheck in the Firebase Console allows Firestore data to be fetched successfully, indicating that the problem is likely due to AppCheck failing.

**Additional Information:**
I've already tried multiple configurations without success. Any insights into potential causes or solutions would be greatly appreciated.

**Comment by himesh-cm** [comment]
👋 @FltSv,

Which version of App Check are you using in your project? Please verify your AppCheck setup in both the Firebase Console and your Flutter code. Make sure that the correct provider (appAttest or deviceCheck) is activated and that all necessary configurations (like the Apple Team ID) are correctly set.

Here is a similar issue thread - https://github.com/firebase/flutterfire/issues/9484

**Reply by FltSv** [reply]
@himesh-cm ,
Thank you for your response.

The version of AppCheck we are using in the project is 0.3.0+4. The AppCheck configuration in our Flutter code is as follows. Is there anything else that needs to be added in the Flutter code?
```
await FirebaseAppCheck.instance.activate(
    androidProvider: AndroidProvider.playIntegrity,
    appleProvider: AppleProvider.deviceCheck,
);
```
In the Firebase Console, we have configured AppCheck for both AppAttest and DeviceCheck. The secret key, key ID, and team ID have all been set up, and everything seems to be configured correctly at first glance.

I also reviewed the thread you mentioned with a similar issue. Although the version we are using is newer, I tried switching to AppleProvider.deviceCheck as a precaution, but unfortunately, the results were the same.

If there are any other aspects we should check, please let me know.

Thank you for your assistance.

**Reply by himesh-cm** [reply]
Did you enable the `App Attest` capability for your bundle ID? The iOS setup explicitly mentions doing so, as shown here - https://firebase.google.com/docs/app-check/ios/app-attest-provider. 

If you haven't done so already, please enable it, resave your provisioning profile, and upload it again on codemagic. Then try again.

![Screen Shot 2024-09-03 at 14 34 12 PM](https://github.com/user-attachments/assets/02111693-0ec0-46da-827b-1ec33994b0b2)

**Reply by FltSv** [reply, marked as answer]
Thank you for the additional information.
I had already enabled the App Attest capability and tried regenerating the provisioning profile, but unfortunately, there was no change.

I was at a loss, but after reviewing various settings, I was able to resolve the issue on my own. It turns out that the `appId` listed in `firebase_options.dart` was the only one that was different from the one in the Firebase Console.

Although I had successfully updated the `<FlutterProject>/ios/Runner/GoogleService-info.plist` and other configuration files, I had overlooked updating `FirebaseOptions` , which caused the problem.

Thank you for your help, and I hope this information will be useful to future visitors.
